Source/Document Structure/SvgUse.cs

```diff
@@ -131,6 +167,19 @@ protected override void Render(ISvgRenderer renderer)
                 var element = this.OwnerDocument.IdManager.GetElementById(this.ReferencedElement) as SvgVisualElement;
                 if (element != null)
                 {
+                    var ew = Width.ToDeviceValue(renderer, UnitRenderingType.Horizontal, this);
+                    var eh = Height.ToDeviceValue(renderer, UnitRenderingType.Vertical, this);
+                    if (ew > 0 && eh > 0)
+                    {
+                        var viewBox = element.Attributes.GetAttribute<SvgViewBox>("viewBox");
+                        if (viewBox!=SvgViewBox.Empty && Math.Abs(ew - viewBox.Width) > float.Epsilon && Math.Abs(eh - viewBox.Height) > float.Epsilon)
+                        {
+                            var sw = ew / viewBox.Width;
+                            var sh = eh / viewBox.Height;
+                            renderer.ScaleTransform(sw, sh, MatrixOrder.Prepend);
+                        }
+                    }
+

```

Comment:
Maybe you could use `ViewBox.AddViewBoxTransform` here instead? Not sure about the aspect ratio, though.

Reply:
I can see the intention of the `ViewBox.AddViewBoxTransform`. If the referenced element was an `SvgFragment`, then I could inject the Width/Height from the `SvgUse` element into it. The problem is that not all of the elements that support viewBox are `SvgFragments`.

I could investigate a more robust approach - but given that noone else has raised an issue - and that the solution works for my specific use-case - could we merge as-is?

Reply:
That's ok - it was only a suggestion, I'm fine with the current state.